mosviz/controls/slit_controller.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+
+from glue.core import HubListener
+
+from regions import RectangleSkyRegion, RectanglePixelRegion, PixCoord
+
+from matplotlib.patches import Rectangle
+
+from astropy.coordinates import Angle, SkyCoord
+from astropy.wcs.utils import proj_plane_pixel_area
+from astropy import units as u
+
+from ..controls.slit_selection_ui import SlitSelectionUI
+
+
+class SlitController(HubListener):
+    """
+    Controller that constructs and stores
+    a rectangular slit.
```

Comment:
I realize the name "slit" implies rectangle, but I don't think it's a good idea to restrict this to rectangles.  Part of the point of using regions is that that allows *any* valid region shape to be used here.

It may make sense to initially only support rectangles since that's the most common slit type, but I think you shouldn't limit it to this. In particular, it's important to know that rectangles on the *sky* are not the same as rectangles in pixel space (they are usually distorted).  Also there exist spectrographs (most notably, COS), where the apertures are circular instead of slits.

More broadly, this means you do *not* want to implement the properties like `length`/`width` here, because they are specific to the rectangle region and would give errors if, say, I used a rectangle.

Reply:
I made the code more generalized for this part.